File: Kotlin/ByTier/A/2151_A_Incremental_Subarray.kt
// Problem: CF 2151 A - Incremental Subarray
// https://codeforces.com/contest/2151/problem/A

/**
 * Code purpose: Given a pattern array `a` and a generated sequence `b` (where b is built by concatenating
 * subsequences [1], [1,2], ..., [1,2,...,n]), this program counts how many times the pattern `a`
 * appears as a subarray in `b`.
 *
 * Algorithm/Techniques: 
 * - Precompute the sequence `b` implicitly using mathematical insight.
 * - For each test case:
 *   1. Parse input values n and m.
 *   2. Read the pattern array `a`.
 *   3. Compute the number of occurrences by checking whether the given subarray can be at the end
 *      of the generated sequence (when it's a full range [1, ..., n]).
 * 
 * Time Complexity: O(m) per test case. The algorithm avoids actually generating or scanning
 *                  the full array `b`, instead analyzing properties of `a` and `n`.
 * Space Complexity: O(m) for storing input array `a`.
 */

fun main() {
    val cin = System.`in`.bufferedReader()
    val cout = System.out.bufferedWriter()
    repeat(cin.readLine().toInt()) {
        val (n, m) = cin.readLine().split(" ").map { it.toInt() }
        val a = cin.readLine().split(" ").map { it.toInt() }
        var ans = 1
        if (a[m-1] - a[0] + 1 == m) {
            // If the range from first to last element of pattern equals its length,
            // then it's a contiguous increasing subsequence that could have started anywhere
            // at the end of the sequence. Count how many such starting positions exist.
            ans = n - a[m-1] + 1
        }
        cout.write("${ans}\n")
    }
    cout.flush()
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/A/2151_A_Incremental_Subarray.py
# Problem: CF 2151 A - Incremental Subarray
# https://codeforces.com/contest/2151/problem/A

"""
Algorithm/Techniques: Simulation, Sorting, Subarray Matching

Time Complexity: O(n^2 + m*log(m)) where n is the maximum number written by James and m is the length of array a.
Space Complexity: O(n^2) for storing the generated array b.

This code simulates the pattern of numbers written by James on the blackboard.
The pattern follows a specific structure:
- First step: 1
- Second step: 1, 2
- Third step: 1, 2, 3
- ...
- nth step: 1, 2, ..., n

So the final array b is formed by concatenating these sequences.
For each test case:
1. Generate the array b of all numbers written by James (simulated).
2. Check if the input array a is sorted and has no duplicates.
3. If so, return n - a[-1] + 1 (number of valid subarray positions).
4. Otherwise, return 1 since we are guaranteed that a is a subarray of b.

The key insight is:
- For an array a = [1, 2, ..., k], if a is sorted and has no duplicates,
  then it will appear as a subarray starting at multiple positions.
  The count depends on the last element of a and how many such sequences begin with that element.
"""

t = int(input())
for _ in range(t):
    n, m = map(int, input().split())
    a = list(map(int, input().split()))
    b = sorted(a)
    # Check if a is already sorted and all elements are unique
    if a == b and len(set(a)) == len(a):
        # If array a forms an increasing sequence from 1 to some k,
        # then number of valid subarray positions = n - last_element + 1
        print(n - a[-1] + 1)
    else:
        # In all other cases, answer is always 1 since we are guaranteed to have at least one match
        print(1)


# https://github.com/VaHiX/CodeForces/